text/0000-multi-part-examples-in-rustdoc/index.md

```diff
@@ -0,0 +1,231 @@
+
+Advantages:
+
+- Seems rather futuristic and perhaps could be part of a larger langauge change, where all comments are by definition markdown, and therefore GUI editors could display rendered versions of comments in place of the actual comments.
+- There's no splitting of code blocks.
+
+We imagine a more convenient authoring experience with the suggested design.
+Even though it feels like a close call, that tips the scale.
+
+Also, the suggested approach is not mutually exclusive with this approach, so that this approach could remain a future possibility.
+
+## Tree variant
+
+This design is similar to the suggested design, but blocks may form tree structures. Multiple blocks may continue from a same prior block.
+
+The advantage is that it would allow code sharing between multiple examples.
+
+The disadvantage is that the result might not be reader-friendly.
+Documentation is pages that are consumed usually top to bottom.
+Adding code blocks that continue one another in some tree structure may seem like a good idea from the perspective of the developer/author but would probably not actually be nice for the reader.
+
+## Interleaved variant
+
+This is a feature that is additional to the suggested design, where the blocks of a multi-part example may interleave with blocks of other multi-part examples.
+
+We feel that this could lead to confusing documentation.
+
+## Q: What is the impact of not doing this?
+
+If this feature is not implemented and no other solution is implemented for including text in long code examples, then long code examples in rust documentation continue to be inconvenient for both authors and readers.
+
+# Prior art
+[prior-art]: #prior-art
+
+This RFC is the evolution of a comment analyzing the existing [RFC 2894][rfc_2894]. It follows [this comment][rfc_comment].
+
+If we look at how programming books are commonly formatted, we will often find multi-part code examples.
+They are split into multiple code blocks and continue each other linearly.
+Between the code blocks is usually verbose educational text.
+Rust documentation authors already seem to follow this style while using one or more of the workarounds described earlier.
+This RFC officialy adopts this style.
+
+One could claim that this RFC allows code examples to be written in [literate programming][literate_programming] style.
+
+We are not familiar with a feature equivalent to the doc-tests in other languages.
+
+# Unresolved questions
+[unresolved-questions]: #unresolved-questions
+
+We have some todos in the spec section.
+Otherwise, none that we could think of.
+
+# Future possibilities
+[future-possibilities]: #future-possibilities
+
+One future possibility is that code editors would support this.
+Assuming, of course, that they would support language features inside of code examples in documentation, at all.
+
+We expect early adoption of this feature.
```

Comment:
What does this mean?

Reply:
In 7e6c746440c2bcf77498a9eae37db8e199d91779 we tried to clarify.
In 524dc6452306d71e5b1c7d1f7ab2b33ab0f3044e we removed a paragraph in which we meant to express our prediction that this feature will be used and even used if it is behind a feature flag.